### PR TITLE
Fixed #31014 -- Added FromWKB and FromWKT GIS functions.

### PR DESCRIPTION
**Pull request by sir-sigurd**
https://code.djangoproject.com/ticket/31014

- [ ] add `srid` parameter?

### CODE REVIEW

File: tests/gis_tests/geoapp/test_functions.py [hunk 1]
```diff
@@ -8,7 +8,7 @@
     GEOSGeometry, LineString, Point, Polygon, fromstr,
 )
 from django.contrib.gis.measure import Area
-from django.db import NotSupportedError, connection
+from django.db import NotSupportedError, connection, models
 from django.db.models import Sum
```
Comment (pope1ni) [inline tests/gis_tests/geoapp/test_functions.py:12]:
```suggestion
from django.db import NotSupportedError, connection
from django.db.models import Sum, Value
```

File: tests/gis_tests/geoapp/test_functions.py [hunk 2]
```diff
@@ -257,6 +257,29 @@ def test_force_polygon_cw(self):
...
+    @skipUnlessDBFeature("has_FromWKB_function")
+    def test_fromwkb(self):
+        g = Point(56.811078, 60.608647)
+        g2 = City.objects.values_list(
+            functions.FromWKB(
+                models.Value(
+                    bytes(g.wkb),
+                    output_field=models.BinaryField(),
+                )
+            ),
```
Comment (pope1ni) [inline tests/gis_tests/geoapp/test_functions.py:269]: As of 1e38f1191de21b6e96736f58df57dfb851a28c1f, `output_field` doesn't need to be defined here.

```suggestion
            functions.FromWKB(Value(g.wkb.tobytes())),
```

File: tests/gis_tests/geoapp/test_functions.py [hunk 3]
```diff
@@ -257,6 +257,29 @@ def test_force_polygon_cw(self):
...
+            flat=True,
+        )[0]
+        self.assertEqual(g, g2)
+
+    @skipUnlessDBFeature("has_FromWKT_function")
+    def test_fromwkt(self):
+        g = Point(56.811078, 60.608647)
+        g2 = City.objects.values_list(
+            functions.FromWKT(models.Value(g.wkt)),
```
Comment (pope1ni) [inline tests/gis_tests/geoapp/test_functions.py:278]:
```suggestion
            functions.FromWKT(Value(g.wkt)),
```

File: docs/ref/contrib/gis/functions.txt [hunk 4]
```diff
@@ -335,6 +331,32 @@ of the polygon/multipolygon in which all exterior rings are oriented clockwise
...
+Oracle, `PostGIS <https://postgis.net/docs/ST_GeomFromWKB.html>`__, SpatiaLite
+
+Creates geometry from `Well-known binary (WKB)`_ representation.
+
+``FromWKT``
+===========
+
+.. class:: FromWKT(expression, **extra)
+
+.. versionadded:: 3.1
```
Comment (pope1ni) [inline docs/ref/contrib/gis/functions.txt:352]:
```suggestion
.. versionadded:: 3.2
```

File: docs/ref/contrib/gis/functions.txt [hunk 5]
```diff
@@ -335,6 +331,32 @@ of the polygon/multipolygon in which all exterior rings are oriented clockwise
 and all interior rings are oriented counterclockwise. Non-polygonal geometries
 are returned unchanged.
 
+``FromWKB``
+===========
+
+.. class:: FromWKB(expression, **extra)
+
+.. versionadded:: 3.1
```
Comment (pope1ni) [inline docs/ref/contrib/gis/functions.txt:339]:
```suggestion
.. versionadded:: 3.2
```

**Review by pope1ni — CHANGES_REQUESTED** [review]
@sir-sigurd Are you interested in picking this up again?

AFAICT, the only things missing are a release note and your outstanding question regarding SRIDs.

**Comment by felixxm** [comment]
Closing due to inactivity.

**Pull request closed without merging**